### DISCUSSION TITLE
Pump S100101 + D1 Mini board - What is wrong?

### DISCUSSION THREAD

**Discussion by PrzemLinek** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#571] [Q&A]
I am using the D1 Mini board (as in this guide https://antanaitis.lt/2022/09/07/wi-fi-on-your-lay-z-spa-make-it-easy-to-remote-control/ )

I connected D1 Mini to the pump (S100101). The console shows me the current temperature. When I press the buttons I hear sounds (different for each button). But nothing else happens. It looks like the panel is locked, but how to unlock it.

I can see the current temperature on the web interface, but I can't do anything else either.

Any suggestions?

**Comment by espenfjo** [comment]
This seems to be the same issue I have, and the leading theory is that we are either unlucky with our D1s, or unlucky with our pump not being able to interpret the 3.3V signals from the D1 all the time it is expecting 5V.

> I am using the D1 Mini board (as in this guide https://antanaitis.lt/2022/09/07/wi-fi-on-your-lay-z-spa-make-it-easy-to-remote-control/ )
> 
> I connected D1 Mini to the pump (S100101). The console shows me the current temperature. When I press the buttons I hear sounds (different for each button). But nothing else happens. It looks like the panel is locked, but how to unlock it.
> 
> I can see the current temperature on the web interface, but I can't do anything else either.
> 
> Any suggestions?

This seems to be the same issue I have, and the leading theory is that we are either unlucky with our D1s, or unlucky with our pump not being able to interpret the 3.3V signals from the D1 all the time it is expecting 5V.

**Reply by PrzemLinek** [reply]
You're probably right. I thought that by supplying D1 with 5V, the IO pins run at 5V, but they still work at 3.3V. 
If the pump requires 5V, I'll check if the logical level shifter will help.

Thanks

**Reply by joelfak** [reply]
I just followed the same guide and I run in to the same problem. It starts up fine, showing the IP address and then the temperature. When pressing the buttons I get different sounds and then the display goes black. Did you solve the problem. Did you try to change the voltage?

**Reply by metaljay** [reply]
It’s because you aren’t using a low level converter, many of us tried the same guide and on newer pumps it just doesn’t work. Easier to just buy your own kit / one assembled

**Reply by joelfak** [reply]
I thought as much. I have tried connecting one of these https://www.amazon.se/dp/B07NNRS8FS?ref_=pe_24982401_513610551_E_301_dt_1&th=1 to overcome the voltage problem, but it did not work out. I am a complete beginner doing this, so maybe I should try a kit instead :)

**Reply by metaljay** [reply]
Depends where you live; but I sell kits. Especially if you can solder yourself it’s cheaper. 

https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/615

**Reply by joelfak** [reply]
Yes, I noticed that thread yesterday, too bad I did not find it at first. I stil send you an email, I live in Sweden.

**Reply by Smedgarden** [reply]
@joelfak Har du hittat en lösning?

Sitter just nu med exakt samma problem, har det löst sig?
